lib/buffer.js

```diff
@@ -446,6 +499,12 @@ var writeWarned = false;
 const writeMsg = '.write(string, encoding, offset, length) is deprecated.' +
                  ' Use write(string[, offset[, length]][, encoding]) instead.';
 Buffer.prototype.write = function(string, offset, length, encoding) {
+  if (!(this instanceof Buffer))
+    throw new TypeError('this is not an instance of Buffer');
+
+  if (typeof string !== 'string')
+    throw new TypeError('Argument must be a string');
```

Comment:
gah! hate that we have such inconsistency in how we handle types. thanks for clarifying.